lib/Service/StorageInfoProvider.php

```diff
@@ -4,13 +4,124 @@
 namespace OCA\MonthlyStatusEmail\Service;
 
 use OCP\IUser;
+use OC\Files\View;
 
 class StorageInfoProvider {
+	/**
+	 * Calculate the disc space for the given path
+	 *
+	 * BEWARE: this requires that Util::setupFS() was called
+	 * already !
+	 *
+	 * @param string $path
+	 * @param \OCP\Files\FileInfo $rootInfo (optional)
+	 * @return array
+	 * @throws \OCP\Files\NotFoundException
+	 */
 	public function getStorageInfo(IUser $user): array {
-		// make sure FS is setup before querying storage related stuff...
 		\OC_Util::setupFS($user->getUID());
 
-		// Handle storage specific events
-		return \OC_Helper::getStorageInfo('/');
+		// return storage info without adding mount points
+		$includeExtStorage = \OC::$server->getSystemConfig()->getValue('quota_include_external_storage', false);
+		$view = new View("/" . $user->getUID() . "/files");
+		$fullPath = $view->getAbsolutePath($path);
+
+		if (!$rootInfo) {
+			$rootInfo = $view->getFileInfo($path, 'ext');
+		}
+		if (!$rootInfo instanceof \OCP\Files\FileInfo) {
+			throw new \OCP\Files\NotFoundException();
+		}
+		$used = $rootInfo->getSize($includeMountPoints);
+		if ($used < 0) {
+			$used = 0;
+		}
+		$quota = \OCP\Files\FileInfo::SPACE_UNLIMITED;
+		$mount = $rootInfo->getMountPoint();
+		$storage = $mount->getStorage();
+		$sourceStorage = $storage;
+		if ($storage->instanceOfStorage('\OCA\Files_Sharing\SharedStorage')) {
+			$includeExtStorage = false;
+			$internalPath = $storage->getUnjailedPath($rootInfo->getInternalPath());
+		} else {
+			$internalPath = $rootInfo->getInternalPath();
+		}
+		if ($includeExtStorage) {
+			if ($storage->instanceOfStorage('\OC\Files\Storage\Home')
+				|| $storage->instanceOfStorage('\OC\Files\ObjectStore\HomeObjectStoreStorage')
+			) {
+				/** @var \OC\Files\Storage\Home $storage */
+				$user = $storage->getUser();
+			} else {
+				$user = \OC::$server->getUserSession()->getUser();
+			}
+			$quota = OC_Util::getUserQuota($user);
+			if ($quota !== \OCP\Files\FileInfo::SPACE_UNLIMITED) {
+				// always get free space / total space from root + mount points
+				return self::getGlobalStorageInfo($quota, $user, $mount);
```

Comment:
this is not gonna work :/ I probably didn't it this, because I didn't test this code with a SharedStorage

Reply:
this code might also be not needed since we only need to know the used storage for the root folder of an user and not an arbitrary path